ui-v2/app/adapters/policy.js

```diff
@@ -0,0 +1,91 @@
+import Adapter, {
+  REQUEST_CREATE,
+  REQUEST_UPDATE,
+  DATACENTER_QUERY_PARAM as API_DATACENTER_KEY,
+} from './application';
+
+import { PRIMARY_KEY, SLUG_KEY } from 'consul-ui/models/policy';
+import { FOREIGN_KEY as DATACENTER_KEY } from 'consul-ui/models/dc';
+import { OK as HTTP_OK } from 'consul-ui/utils/http/status';
+import { PUT as HTTP_PUT } from 'consul-ui/utils/http/method';
+
+export default Adapter.extend({
+  // _ajaxRequest(options) {
+  //   const hash = Object.assign(
+  //     {},
+  //     options,
+  //     {
+  //       success: function(payload, textStatus, jqXHR) {
+  //         setTimeout(
+  //           () => {
+  //             return options.success(...arguments);
+  //           },
+  //           2000
+
+  //         );
+  //       }
+  //     }
+  //   );
+  //   $.ajax(hash);
+  // },
+  urlForQuery: function(query, modelName) {
+    return this.appendURL('acl/policies', [], this.cleanQuery(query));
+  },
+  urlForQueryRecord: function(query, modelName) {
+    if (typeof query.id === 'undefined') {
+      throw new Error('You must specify an id');
+    }
+    return this.appendURL('acl/policy', [query.id], this.cleanQuery(query));
+  },
+  urlForCreateRecord: function(modelName, snapshot) {
+    return this.appendURL('acl/policy', [], {
+      [API_DATACENTER_KEY]: snapshot.attr(DATACENTER_KEY),
+    });
+  },
+  urlForUpdateRecord: function(id, modelName, snapshot) {
+    return this.appendURL('acl/policy', [snapshot.attr(SLUG_KEY)], {
+      [API_DATACENTER_KEY]: snapshot.attr(DATACENTER_KEY),
+    });
+  },
+  urlForDeleteRecord: function(id, modelName, snapshot) {
+    return this.appendURL('acl/policy', [snapshot.attr(SLUG_KEY)], {
+      [API_DATACENTER_KEY]: snapshot.attr(DATACENTER_KEY),
+    });
+  },
+  urlForTranslateRecord: function(modelName, snapshot) {
+    return this.appendURL('acl/policy/translate', [], {});
+  },
+  dataForRequest: function(params) {
+    const data = this._super(...arguments);
+    switch (params.requestType) {
+      case REQUEST_UPDATE:
+      case REQUEST_CREATE:
+        return data.policy;
+    }
+    return data;
+  },
+  handleResponse: function(status, headers, payload, requestData) {
+    let response = payload;
+    if (status === HTTP_OK) {
+      const url = this.parseURL(requestData.url);
+      switch (true) {
```

Comment:
Yeah it follows the rest of the app, this one is simpler but the others in the rest of the app aren't so simple. I'd rather follow the same pattern and be consistent. 'Generally' I'll use a `switch` instead of lots of `if..else if`s whereas if its just an `if ...else` I'll generally keep with an `if`.